src/test/java/org/openrewrite/java/migrate/RemovedToolProviderConstructorTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class RemovedToolProviderConstructorTest  implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.expectedCyclesThatMakeChanges(2).recipe(new RemovedToolProviderConstructor());
```

Comment:
Would it be possible to write the recipe such that it does not require additional cycles? Usually that's a case of an early return when the target state is already reached.
```suggestion
        spec.recipe(new RemovedToolProviderConstructor());
```

Reply:
Will take a look into doing this